### ISSUE TITLE
Anonymous mode

### ISSUE THREAD

**Issue by BartoszJarocki** [ParabolInc/parabol#9247]
## Issue - Enhancement

Currently, we store the `authorId` even if user creates an anonymous comment. GQL resolvers are ultimately responsible for _not_ exposing author information when the comment was anonymous. This is not ideal and could lead to bugs like #9197 #9196. 

Instead, when comment/reaction is anonymous we should _not_ store any information about author.

### Acceptance Criteria (optional)
- [ ] identify all the places where `authorId` has to be removed for example comments, reactions
- [ ] make sure the proposed is backward compatible ie. previous anonymous comments stay anonymous etc.
- [ ] propose the solution before starting the implementation by writing some sort of technical spec

**Comment by mattkrick** [comment]
I think this may be dangerous.
Let's say I write something anonymous like "I think Bartosz is a jerk". Then, I come to my senses & realize I overreacted. I want to delete my anonymous comment... but i can't because I don't own the comment.

**Comment by Dschoordsch** [comment]
For notifications we can safely set the author to `null` because it links to the comments. For reflections and comments I think it would be best to not allow endless editing, but after a phase ends, set all the anonymous authors to `null`.
I personally would not like to have a situation where we can write `Georg is a jerk` anonymously and then a day later just delete it.
If I accidentally posted something private like a password or similar, I can understand the need for this, but that should be a short time window.

**Comment by mattkrick** [comment]
immutability is _real_ nice & i love the extra privacy, plus I'm all for calling something "done"!
that said, if there's anything i've learned as we've built this, it's that users LOVE hitting undo. We've caved in some cases, like resetting a meeting to an earlier phase. We've compromised in other cases, like adding extra friction when ending a meeting instead of letting them un-end it. if we remove authorId, we're taking functionality away, so the thing we get for it better be GOOD. On a more serious note, this is primary data that no one else in the world has. We've gotta be really cautious with it, but if used responsibly we could answer really interesting questions with the data. Not saying we will, but if we remove the ability to, we're decreasing the value of our data
